src/main/java/org/folio/rest/impl/BatchVoucherHelper.java

```diff
@@ -0,0 +1,94 @@
+package org.folio.rest.impl;
+
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static javax.ws.rs.core.MediaType.APPLICATION_XML;
+import static org.folio.invoices.utils.HelperUtils.handleGetRequest;
+import static org.folio.invoices.utils.ResourcePathResolver.BATCH_VOUCHER_STORAGE;
+import static org.folio.invoices.utils.ResourcePathResolver.resourceByIdPath;
+
+import java.util.Collections;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.Response;
+
+import org.folio.HttpStatus;
+import org.folio.jaxb.XMLConverter;
+import org.folio.rest.jaxrs.model.BatchVoucher;
+import org.folio.rest.jaxrs.model.Error;
+import org.folio.rest.jaxrs.model.Errors;
+import org.folio.rest.jaxrs.model.jaxb.BatchVoucherType;
+import org.folio.spring.SpringContextUtil;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.core.convert.ConversionService;
+
+import io.vertx.core.Context;
+import io.vertx.core.Vertx;
+import io.vertx.core.json.JsonObject;
+import me.escoffier.vertx.completablefuture.VertxCompletableFuture;
+
+public class BatchVoucherHelper extends AbstractHelper {
+  private static final String HEADER_ERROR_MSG = "Accept header must be [\"application/xml\",\"application/json\"]";
+
+  @Autowired
+  private XMLConverter xmlConverter;
+  @Autowired
+  private ConversionService conversionService;
+
+  public BatchVoucherHelper(Map<String, String> okapiHeaders, Context ctx, String lang) {
+    super(okapiHeaders, ctx, lang);
+    SpringContextUtil.autowireDependencies(this, Vertx.currentContext());
+  }
+
+  /**
+   * Gets batch voucher by id
+   *
+   * @param id batch voucher uuid
+   * @return completable future with {@link BatchVoucher} on success or an exception if processing fails
+   */
+  public CompletableFuture<Response> getBatchVoucherById(String id, String acceptHeader) {
+    CompletableFuture<Response> future = new VertxCompletableFuture<>(ctx);
+    String endpoint = resourceByIdPath(BATCH_VOUCHER_STORAGE, id, lang);
+    handleGetRequest(endpoint, httpClient, ctx, okapiHeaders, logger)
+      .thenApplyAsync(jsonObject -> buildSuccessResponse(jsonObject, acceptHeader))
+      .thenAccept(future::complete)
+      .exceptionally(t -> {
+        future.completeExceptionally(t);
+        return null;
+      });
+    return future;
+  }
+
+  private Response buildSuccessResponse(JsonObject jsonObjectBatchVoucher, String acceptHeader) {
+    BatchVoucher jsonBatchVoucher = jsonObjectBatchVoucher.mapTo(BatchVoucher.class);
+    switch (acceptHeader) {
+    case APPLICATION_XML:
+      BatchVoucherType xmlBatchVoucher = conversionService.convert(jsonBatchVoucher, BatchVoucherType.class);
```

Comment:
@AndreiMPV @craigmcnally ,
looks like this solution is very complex& I just tried to use Jackson here:
`    ObjectMapper objectMapper = new ObjectMapper();
        ObjectMapper xmlMapper = new XmlMapper();
        JsonNode tree = objectMapper.readTree(jsonObjectBatchVoucher.encode());
        String jsonAsXml = xmlMapper.writer().withRootName("atchVoucher").writeValueAsString(tree);
        return buildSuccessResponse(jsonAsXml, APPLICATION_XML);`
instead of
`BatchVoucherType xmlBatchVoucher = conversionService.convert(jsonBatchVoucher, BatchVoucherType.class);
        String xmlBatchVoucherResponse = xmlConverter.marshal(xmlBatchVoucher, true);`
and unit tests passed successfully.

Reply:
@khandramai @craigmcnally
Thanks for pointing. 
I agree with you and also not so happy with a bunch of Converters and codes around that.
There are alternative approaches how to get XML from response:
1. I also tried to implement this functionality using the method that Slava advised, but failed to achieve the correct collection conversion (List<BatchedVoucher>) and also in the conversion of DateTime types.
2. There was a other solution, which basen  on **org.folio.rest.tools.plugins.ResourceMethodExtensionPlugin**, which able to update Java sources and add annotations to existing models. In this case, we could use the scheme only for validation. Unfortunately, I was not able to implement this approach, some problems  were occurred in the implementation.
3. At the moment, it was partially possible, using reflection, to annotate the JSON models with JAXB anotations and this makes it possible to eliminate the needed to generate separate models according to the XSD scheme and at the same time use the validation of the produced XML according to the XSD scheme.
With such a decision, we can 
a) still use JAXBMarshaller for the production of XML(OOTB used schema validation)
b) or use Jackson XmlMapperr
c) or use JAXB utility class.
For me the best to use JAXBMarshaller.
Thanks,
Andrei

Reply:
We use JaxbMarshaller in both mod-gobi and mod-oai-pmh, so there's some precedence there.  I would say that schema validation is strongly desired, if not a hard requirement.  If that's not possible (or possible but difficult) with the other approaches the decision is clear.

That said, I'm not sure I fully understand the need to use reflection - we didn't run into that in either of the aforementioned modules.  What's different here that necessitates this extra complexity?

Reply:
Also note that there's precedence for using jackson in edge-rtac.  Here we created the POJOs ourselves with the necessary annotations.  IIRC we generated the POJOs from the schema once, copied them into the main codebase and augmented them with necessary annotations/code.

Reply:
BTW... https://github.com/folio-org/mod-oai-pmh/blob/795d0c72b898e43688f26da152225b02ff0d91bd/src/main/java/org/folio/oaipmh/ResponseHelper.java#L30

D'oh!